api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java

```diff
@@ -241,6 +249,36 @@ void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForDwpHouseholdIde
         wiremockManager.assertThatEligibilityRequestMade();
     }
 
+    @Test
+    void shouldReturnDuplicateStatusWhenTwoEligibleClaimsForSameNino() throws JsonProcessingException, InterruptedException {
+        //Given
+        NewClaimDTO dto = aValidClaimDTO();
+        CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse
+                = anIdMatchedEligibilityConfirmedUCResponseWithAllMatchesAndDwpHouseIdentifier(DWP_HOUSEHOLD_IDENTIFIER);
+        wiremockManager.stubEligibilityResponse(identityAndEligibilityResponse);
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+
+        //When
+        List<Future<Object>> futures = executor.invokeAll(List.of(
+                () -> restTemplate.exchange(buildCreateClaimRequestEntity(dto), ClaimResultDTO.class),
+                () -> restTemplate.exchange(buildCreateClaimRequestEntity(dto), ClaimResultDTO.class)));
+
+        List<ResponseEntity<ClaimResultDTO>> results = futures.stream().map(objectFuture -> {
+            try {
+                return (ResponseEntity<ClaimResultDTO>) objectFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }).collect(Collectors.toList());
+
+        //Then
+        //one transaction will succeed because it commits first.
+        assertThat(results.stream().anyMatch(r -> r.getBody().getClaimStatus() == ClaimStatus.NEW)).isTrue();
+
+        //one transaction will fail causing error because there is already an active claim with same nino.
+        assertThat(results.stream().anyMatch(r -> r.getStatusCode() == HttpStatus.INTERNAL_SERVER_ERROR)).isTrue();
```

Comment:
Similarly here you need to specifically check that you have INTERNAL_SERVER_ERROR and OK, this only checks that at least one status code is INTERNAL_SERVER_ERROR.

Reply:
Just to note that the other status will be CREATED, not OK.